test/e2e/soak_k8petstore.go

```diff
@@ -0,0 +1,181 @@
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	//"path"
+	"path/filepath"
+	"strconv"
+	"syscall"
+	"time"
+
+	//"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	k8bpsContainerVersion = "r.2.8.19" // Container version, see the examples/k8petstore dockerfiles for details.
+	k8bpsThroughputDummy  = "0"        // Polling time = 0, since we poll in ginkgo rather than using the shell script tests.
+	k8bpsRedisSlaves      = "1"        // Number of redis slaves.
+	k8bpsDontRunTest      = "0"        // Don't bother embedded test.
+)
+
+// readTransactions reads # of transactions from the k8petstore web server endpoint.
+// for more details see the source of the k8petstore web server.
+
+func readTransactions(c *unversioned.Client, ns string) (error, int) {
+	body, err := c.Get().
+		//TODO, need to fix this to work
+		//with node ports
+		Namespace(ns).
+		Prefix("proxy").
+		Resource("services").
+		Name("frontend").
+		Suffix("llen").
+		DoRaw()
+	if err != nil {
+		return err, -1
+	} else {
+		totalTrans, err := strconv.Atoi(string(body))
+		return err, totalTrans
+	}
+}
+
+// runK8petstore runs the k8petstore application, bound to external nodeport, and
+// polls until minExpected transactions are acquired, in a maximum of maxSeconds.
+func runK8petstore(restServers int, loadGenerators int, c *unversioned.Client, ns string, minExpected int, maxSeconds int) {
+
+	var err error = nil
+	k8bpsScriptLocation := filepath.Join(testContext.RepoRoot, "examples/k8petstore/k8petstore-nodeport.sh")
+
+	//Get the count of minions.  We'll use this to decide expected throughput and loadgen/REST server count.
+	cmd := exec.Command(
+		k8bpsScriptLocation,
+		"kubectl", // For dev, replace w/ "cluster/kubectl.sh"
+		k8bpsContainerVersion,
+		k8bpsThroughputDummy, // Throughput measurement time.  We send 0 here since we're not running tests at all.
+		strconv.Itoa(restServers),
+		strconv.Itoa(loadGenerators),
+		k8bpsRedisSlaves,
+		k8bpsDontRunTest, // Don't bother embedded test.
+		ns,
+	)
+
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	Logf("Starting k8petstore application....")
+	//Run the k8petstore app, and log / fail if it returns any errors.
+	//This should return quickly, assuming containers are downloaded.
+	if err = cmd.Start(); err != nil {
+		log.Fatal(err)
+	}
+	//Make sure exit code != 0
+	if err = cmd.Wait(); err != nil {
+		if exiterr, ok := err.(*exec.ExitError); ok {
+			if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+				log.Printf("Exit Status: %d", status.ExitStatus())
+			}
+		}
+	}
+	Expect(err).NotTo(HaveOccurred())
+	Logf("... Done starting k8petstore ")
+
+	//Lets poll until we reach min_expected transactions
+	totalTransactions := 0
+	startupTimeoutSeconds := 500
```

Comment:
This is almost 10 minutes, and we want the test to complete in 1 minute.  Can we reduce this?